plugins/DDG4EDM4hep/edm4hep/Geant4Output2EDM4hep.cpp

```diff
@@ -0,0 +1,566 @@
+      for(std::map<std::string, std::string >::const_iterator iter = parameters.begin(), endIter = parameters.end() ; iter != endIter ; ++iter)  {
+	printout( WARNING, "Geant4Output2EDM4hep","saveEventParameters(): not implemented in EDM4hep - not written:   %s  -> %s ",  iter->first.c_str(),iter->second.c_str() ) ;
+
+//        event->parameters().setValue(iter->first,iter->second);
+      }
+    }
+
+  }    // End namespace sim
+}      // End namespace dd4hep
+#endif // DD4HEP_DDG4_GEANT4OUTPUT2EDM4hep_H
+
+//==========================================================================
+//  AIDA Detector description implementation 
+//--------------------------------------------------------------------------
+// Copyright (C) Organisation europeenne pour la Recherche nucleaire (CERN)
+// All rights reserved.
+//
+// For the licensing terms see $DD4hepINSTALL/LICENSE.
+// For the list of contributors see $DD4hepINSTALL/doc/CREDITS.
+//
+// Author     : F.Gaede, DESY
+//
+//==========================================================================
+
+// Framework include files
+#include "DD4hep/InstanceCount.h"
+#include "DD4hep/Detector.h"
+#include "DDG4/Geant4HitCollection.h"
+#include "DDG4/Geant4DataConversion.h"
+#include "DDG4/Geant4Context.h"
+#include "DDG4/Geant4Particle.h"
+#include "DDG4/Geant4Data.h"
+#include "DDG4/Geant4Action.h"
+
+//#include "DDG4/Geant4Output2EDM4hep.h"
+#include "G4ParticleDefinition.hh"
+#include "G4VProcess.hh"
+#include "G4Event.hh"
+#include "G4Run.hh"
+
+
+using namespace dd4hep::sim;
+using namespace dd4hep;
+using namespace std;
+namespace {
+  G4Mutex action_mutex=G4MUTEX_INITIALIZER;
+}
+
+#include "DDG4/Factories.h"
+DECLARE_GEANT4ACTION(Geant4Output2EDM4hep)
+
+/// Standard constructor
+Geant4Output2EDM4hep::Geant4Output2EDM4hep(Geant4Context* ctxt, const string& nam)
+: Geant4OutputAction(ctxt,nam), m_store(0), m_file(0), m_runNo(0), m_runNumberOffset(0), m_eventNumberOffset(0)
+{
+  declareProperty("RunHeader", m_runHeader);
+  declareProperty("EventParametersInt",    m_eventParametersInt);
+  declareProperty("EventParametersFloat",  m_eventParametersFloat);
+  declareProperty("EventParametersString", m_eventParametersString);
+  declareProperty("RunNumberOffset", m_runNumberOffset);
+  declareProperty("EventNumberOffset", m_eventNumberOffset);
+  printout( INFO, "Geant4Output2EDM4hep" ," instantiated ..." ) ;
+  InstanceCount::increment(this);
+}
+
+/// Default destructor
+Geant4Output2EDM4hep::~Geant4Output2EDM4hep()  {
+  G4AutoLock protection_lock(&action_mutex);
+  if ( m_file )  {
+    m_file->finish();
+    detail::deletePtr(m_file);
+  }
+  InstanceCount::decrement(this);
+}
+
+// Callback to store the Geant4 run information
+void Geant4Output2EDM4hep::beginRun(const G4Run* run)  {
+  if ( 0 == m_file && !m_output.empty() )   {
+    G4AutoLock protection_lock(&action_mutex);
+    m_store = new podio::EventStore ;
+    m_file = new podio::ROOTWriter(m_output, m_store);
```

Comment:
Does the ROOTWriter own the m_store after this call?  The m_stire doesn't get cleaned up in the d'tor.

Reply:
No it doesn't. I added a `delete` in the destructor.